0_intro/live_coding/yes_3/src/main.rs

```diff
@@ -0,0 +1,52 @@
+// Show that you can use multiple modules at once
+use clap::{Arg, App};
+
+// Probably need to change this from Option<String> to Option<&str>
```

Comment:
I don't understand this

Reply:
In `yes_2` and `yes_1` the argument is an `Option<String>`, because `std::env::args()` returns an iterator over Strings, not &strs. Clap, in the other hand, returns a &str when something matches the argument parser. So to go from `std::env::args()` to `clap` you'll need to change the type of this parameter. Clap's `value_of` actually return an `Option<&str>` depending on whether the argument was passed or not so it works out pretty well

Reply:
Ok. The ``probably`` is misleading imao

Reply:
Would something like
```suggestion
// If you were using an Option<String> as parameter, you will need to change it to Option<&str>
```
be better?

This way, the whole comment would become
```
// If you were using an Option<String> as parameter, you will need to change it to Option<&str>
// since std::env::args().nth(...) returns a String but clap will return an &str.
// You can also get started on the differences between &str and String, and add an analogy
// to C/C++
```

Reply:
> Would something like (....) be better ?

Yes, perfect